smarts/core/smarts.py

```diff
@@ -661,7 +661,7 @@ def _reset_providers(self):
         for provider in self.providers:
             provider.reset()
 
-    def _step_providers(self, actions, dt) -> List[VehicleState]:
+    def _step_providers(self, actions, dt, ego_agent_actions) -> List[VehicleState]:
```

Comment:
This is a bit unwieldy, we are asking for both all the actions and then, further, some of the actions. Perhaps send in only the keys of the active agents?

The method may be the wrong place to perform validation, the agents can be validated a method a level up as the last part of step 2. Only because the warnings are useful rather than critical.

Reply:
Very true indeed